src/tallies/tally_scoring.cpp

```diff
@@ -1271,6 +1275,11 @@ score_general_ce(Particle& p, int i_tally, int start_index, int filter_index,
     case N_GAMMA:
     case N_P:
     case N_A:
+      // This case block only works if cross sections for these reactions have
+      // been precalculated. When they are not, we revert to the default case,
+      // which looks up cross sections
+      if (!simulation::need_depletion_rx) goto default_case;
```

Comment:
No doubt, using a `goto` is nothing to be proud of. I'm not sure that the alternative is better however. Letting this case fallthrough to the default case might be risky because it's entirely possible that later on someone adds another `case:` before the default block that messes up the fallthrough. I would argue that what I have is more readable (and maintainable) than the fallthrough alternative. Or said another way, explicit (a goto statement) is better than implicit (fallthrough). Honestly, there are worse sins committed in this almost 1000 line function...

Reply:
Haha fair enough - just curious. Sounds like this section of the code could use a little attention at some point as a separate endeavor.

Reply:
Absolutely, it could use some love! :heart: